src/algorithms/pid/MergeParticleID.cc

```diff
@@ -0,0 +1,199 @@
+    }
+  }
+
+  // trace logging
+  if(m_log->level() <= spdlog::level::trace) {
+    m_log->trace("{:-<70}","Resulting `particle_pids` ");
+    for(auto& [id_particle, idx_paired_list] : particle_pids) {
+      m_log->trace("id_particle={}", id_particle);
+      for(auto& [idx_coll, idx_pid] : idx_paired_list)
+        m_log->trace("  (idx_coll, idx_pid) = ({}, {})", idx_coll, idx_pid);
+    }
+  }
+
+  // --------------------------------------------------------------------------------
+
+
+  // loop over charged particles, combine weights from the associated `CherenkovParticleID` objects
+  // and create a merged output `CherenkovParticleID` object
+  m_log->trace("{:-<70}","Begin Merging PID Objects ");
+  for(auto& [id_particle, idx_paired_list] : particle_pids) {
+
+    // trace logging
+    m_log->trace("Charged Particle:");
+    m_log->trace("  id = {}", id_particle);
+    m_log->trace("  PID Hypotheses:");
+
+    // create mutable output `CherenkovParticleID` object `out_pid`
+    auto out_pid = out_pids->create();
+    out_pid.setNpe(0.0);
+    out_pid.setRefractiveIndex(0.0);
+    out_pid.setPhotonEnergy(0.0);
+
+    // define `pdg_2_out_hyp`: map of PDG => merged output hypothesis
+    std::unordered_map< decltype(edm4eic::CherenkovParticleIDHypothesis::PDG), edm4eic::CherenkovParticleIDHypothesis > pdg_2_out_hyp;
+
+    // merge each input `CherenkovParticleID` objects associated with this charged particle
+    for(auto& [idx_coll, idx_pid] : idx_paired_list) {
+      const auto& in_pid = in_pid_collections_list.at(idx_coll)->at(idx_pid);
+
+      // logging
+      m_log->trace("    Hypotheses for PID result (idx_coll, idx_pid) = ({}, {}):", idx_coll, idx_pid);
+      Tools::PrintHypothesisTableHead(m_log,6);
+
+      // merge scalar members
+      out_pid.setNpe(out_pid.getNpe() + in_pid.getNpe()); // sum
+      out_pid.setRefractiveIndex( out_pid.getRefractiveIndex() + in_pid.getNpe() * in_pid.getRefractiveIndex() ); // NPE-weighted average
+      out_pid.setPhotonEnergy(    out_pid.getPhotonEnergy()    + in_pid.getNpe() * in_pid.getPhotonEnergy()    ); // NPE-weighted average
+
+      // merge photon Cherenkov angles
+      for(auto in_photon_vec : in_pid.getThetaPhiPhotons())
+        out_pid.addToThetaPhiPhotons(in_photon_vec);
+
+      // relate the charged particle
+      if(!out_pid.getChargedParticle().isAvailable()) // only needs to be done once
+        out_pid.setChargedParticle(in_pid.getChargedParticle());
+
+      // merge PDG hypotheses, combining their weights and other members
+      for(auto in_hyp : in_pid.getHypotheses()) {
+        Tools::PrintHypothesisTableLine(m_log,in_hyp,6);
+        auto out_hyp_it = pdg_2_out_hyp.find(in_hyp.PDG);
+        if(out_hyp_it == pdg_2_out_hyp.end()) {
+          edm4eic::CherenkovParticleIDHypothesis out_hyp;
+          out_hyp.PDG    = in_hyp.PDG; // FIXME: no copy constructor?
+          out_hyp.npe    = in_hyp.npe;
+          out_hyp.weight = in_hyp.weight;
```

Comment:
Not using setters here?

Reply:
`edm4eic::CherenkovParticleIDHypothesis` is a `component` (no setters).

Reply:
i.e., not a `datatype`, which has setters.